File: app/src/main/java/com/radio/app/models/Transcript.java
package com.radio.app.models;

import java.io.Serializable;

public class Transcript implements Serializable {
    private String episodeId;
    private long segmentStart;
    private long segmentEnd;
    private String text;
    private double confidence;

    public Transcript() {}

    public Transcript(String episodeId, long segmentStart, long segmentEnd, String text) {
        this.episodeId = episodeId;
        this.segmentStart = segmentStart;
        this.segmentEnd = segmentEnd;
        this.text = text;
    }

    public String getEpisodeId() { return episodeId; }
    public void setEpisodeId(String episodeId) { this.episodeId = episodeId; }
    public long getSegmentStart() { return segmentStart; }
    public void setSegmentStart(long segmentStart) { this.segmentStart = segmentStart; }
    public long getSegmentEnd() { return segmentEnd; }
    public void setSegmentEnd(long segmentEnd) { this.segmentEnd = segmentEnd; }
    public long getStartTime() { return segmentStart; }
    public void setStartTime(long startTime) { this.segmentStart = startTime; }
    public long getEndTime() { return segmentEnd; }
    public void setEndTime(long endTime) { this.segmentEnd = endTime; }
    public String getText() { return text; }
    public void setText(String text) { this.text = text; }
    public double getConfidence() { return confidence; }
    public void setConfidence(double confidence) { this.confidence = confidence; }
}

File: app/src/main/java/com/radio/app/models/Transcript.kt
package com.radio.app.models

import java.io.Serializable

data class Transcript(
    var episodeId: String? = null,
    var segmentStart: Long = 0L,
    var segmentEnd: Long = 0L,
    var text: String? = null,
    var confidence: Double = 0.0
) : Serializable {

    var startTime: Long
        get() = segmentStart
        set(value) { segmentStart = value }

    var endTime: Long
        get() = segmentEnd
        set(value) { segmentEnd = value }
}
